### PR TITLE
Base for typescript migration

### PR DESCRIPTION
**Pull request by slinkardbrandon**

### CODE REVIEW

**Review by AdrieanKhisbe — CHANGES_REQUESTED** [review]
We need **semi-columns**.
Besides the original code had them.

(But maybe that can wait prettier to be fixed)

**Comment by slinkardbrandon** [comment]
In PR #2 with the prettier rules we can set  semicolons to true

**Comment by hongaar** [comment]
> We need **semi-columns**.

Who needs semi-columns? 😉

**Comment by hongaar** [comment]
Do we want to merge this to `develop` instead? Or maybe create a `next` branch so we keep the 'default' branches (develop/master) in sync with upstream? I'd say let's merge this as a start, and then start working our way through the code / fix ts warnings and errors next.

**Review by hongaar — APPROVED** [review]

**Comment by Ore4444** [comment]
@hongaar I think we can use "2.0-ts" as our "develop" branch.
@dthree will merge "2.0-ts" into "master" when we're done.
So for now we can branch out of and open PRs to "2.0-ts".
WDYT?

**Comment by slinkardbrandon** [comment]
We were originally talking about following some resemblance of git flow here and utilizing develop / master and dropping the `2.0-ts` branch from our org (as it will all be TypeScript soon enough). I think in the end we'll create a PR from our org's master to the original vorpal's `2.0-ts` branch. 

It doesn't matter all that much which branch we use but I think @AdrieanKhisbe already set up develop as the default branch

**Comment by AdrieanKhisbe** [comment]
@vorpaljs-reforged/vorpaldevelopers I think it's better we iterate on a `develop` branch, rather to wait for a long time to have `2.0-ts` land. 

(I intend to make some README modifications to make it clear about the vocation of this fork, and make it point on its CI)

**Comment by slinkardbrandon** [comment]
Semi-colons aside (as we can bikeshed about that later), am I good to merge this one?

**Comment by Ore4444** [comment]
I agree about master/develop.
Can merge this PR.
Personally, I'm not using semicolons when I write code, but I use prettier to add them later.

File: package.json [hunk 1]
```diff
@@ -41,17 +41,23 @@
     "gulp-babel": "^6.1.2",
     "gulp-changed": "^1.3.0",
     "gulp-eslint": "^1.1.1",
+    "gulp-typescript": "^5.0.0-alpha.3",
```
Comment (AdrieanKhisbe) [inline package.json:44]: why picking the alpha?

File: package.json [hunk 2]
```diff
@@ -41,17 +41,23 @@
...
+    "npm-check": "^5.9.0",
     "request": "^2.58.0",
     "should": "^6.0.3",
+    "typescript": "^3.1.3",
     "vorpal-less": "0.0.4",
     "vorpal-repl": "^1.1.8",
     "xo": "^0.9.0"
   },
   "dependencies": {
+    "@types/lodash": "^4.14.117",
```
Comment (AdrieanKhisbe) [inline package.json:58]: @types should be in devDependecies if I'm not mistaken

**Review by AdrieanKhisbe — CHANGES_REQUESTED** [review]
I went over the modifications. To me
This could work as a base for migration.
Though we need to set up lint, and really refactor it. 

However I'll say we should take care of the two comment

**Comment by hongaar** [comment]
I'm going to merge it now so we keep some momentum. Will fix the issues @AdrieanKhisbe brought up in a separate PR.

**Pull request merged**